### PR TITLE
Create a TSP Data Scenario

### PR DESCRIPTION
**Pull request by chrisgilmerproj**
## Description

A couple of my PRs have needed an easy way to generate data.  Then recently I had to blow away my DB for a migration and suddenly realized that bespoke data scenarios are terrible.  This PR addresses that.

## Reviewer Notes

Two things going on here of note:
- New scenario 7 that generates tsp users, shipments, connected shipment offers (and sets some to AWARDED)
- Fixed up testdatagen to use Assertions with the TSP model.

## Setup

First run this code:

```sh
make db_dev_reset db_dev_migrate && go run cmd/generate_test_data/main.go -scenario 7
```

Then start up your server and tsp client and go to http://tsplocal:3000/devlocal-auth/login.
You should now see two Leo Spaceman entries.  Click on one and you should be taken to a queue that has some number of awarded shipments.

## Code Review Verification Steps

* [x] Request review from a member of a different team.

### CODE REVIEW

File: pkg/testdatagen/make_shipment_offer_records.go [hunk 1]
```diff
@@ -103,9 +103,13 @@ func CreateShipmentOfferData(db *pop.Connection, numTspUsers int, numShipments i
 
 	// Create TSP Users
 	for i := 1; i <= numTspUsers; i++ {
+		email := fmt.Sprintf("leo_spaceman%d@example.com", i)
 		tspUserAssertions := Assertions{
+			User: models.User{
+				LoginGovEmail: email,
+			},
```
Comment (chrisgilmerproj) [inline pkg/testdatagen/make_shipment_offer_records.go]: Here's the magic for displaying the correct email on the local login page.

File: pkg/testdatagen/make_shipment_offer_records.go [hunk 2]
```diff
@@ -141,6 +146,7 @@ func CreateShipmentOfferData(db *pop.Connection, numTspUsers int, numShipments i
 				SourceGBLOC:             &sourceGBLOC,
 				Market:                  &market,
 				Move:                    &move,
+				Status:                  shipmentStatuses[rand.Intn(len(shipmentStatuses))],
```
Comment (chrisgilmerproj) [inline pkg/testdatagen/make_shipment_offer_records.go]: Here's where I randomly set the status.

**Comment by chrisgilmerproj** [comment]
Looks like I have a failing test.  I'll try to address that in the morning.

File: pkg/testdatagen/make_shipment_offer_records.go (hunk 1 above)
Comment (kilbergr) [reply pkg/testdatagen/make_shipment_offer_records.go]: 👍

**Comment by chrisgilmerproj** [comment]
Thanks @kilbergr!  Let's see if I can answer your questions:

| Why are there a ton of other tsp users who give me an "Unauthorized" page when I select them? Is that from the test data generator? I wonder whether that could be changed BUT IT'S A VERY VERY LOW PRIORITY.

I don't know.  I was looking into this last night but I don't think its related to my code change.  I see this all the time and its somewhat annoying to get another 42ish users in the system when I didn't ask for them.

| This doesn't have a pivotal task attached. It's probably a matter of you doing auxiliary to your work because you need it, but I strongly recommend writing a task for anything you work on to help your planning process. Will especially come in handy as new folks join your team.

Good point, I should try to keep these all linked in Pivotal.  I'll check that my other PRs are all linked.

**Pull request merged**